lightning/src/chain/channelmonitor.rs

```diff
@@ -1796,12 +1796,15 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		} else if self.latest_update_id + 1 != updates.update_id {
 			panic!("Attempted to apply ChannelMonitorUpdates out of order, check the update_id before passing an update to update_monitor!");
 		}
+		let mut ret = Ok(());
 		for update in updates.updates.iter() {
 			match update {
 				ChannelMonitorUpdateStep::LatestHolderCommitmentTXInfo { commitment_tx, htlc_outputs } => {
 					log_trace!(logger, "Updating ChannelMonitor with latest holder commitment transaction info");
 					if self.lockdown_from_offchain { panic!(); }
-					self.provide_latest_holder_commitment_tx(commitment_tx.clone(), htlc_outputs.clone())?
+					if let Err(e) = self.provide_latest_holder_commitment_tx(commitment_tx.clone(), htlc_outputs.clone()) {
+						ret = Err(e);
```

Comment:
Hmm, not sure they care? The errors are all handled the same at the callsite, the only difference is the human-readable part of the error message, which is ultimately just logged anyway? Would you prefer if the error type was `()` and we drop the reason logging at the callsite?

Reply:
Was thinking more if someone was not using `ChainMonitor` they may want access to the error messages. Not sure if that would be in a watchtower or other custom scenario. If you don't think it would be used in any custom setups, then yeah might as well drop it from the interface rather than giving back a partial error.

Reply:
Right, if we wanted the error messages to be used somehow we'd want to move them to some kind of enum describing what to do in different cases. Users definitely shouldn't try to match the human-readable error string and try to change behavior based on it. I'll drop the text and make the error type a `()`

Reply:
Done!